src/service/reports.js

```diff
@@ -0,0 +1,75 @@
+const { userModel, reportModel } = require("../db/mongo");
+const logger = require("../modules/logger");
+
+const reportPopulateOption = [
+  {
+    path: "creatorId",
+    select: "-_id user",
+    populate: { path: "user", select: "_id id name nickname profileImageUrl" },
+  },
+  {
+    path: "reportedId",
+    select: "-_id user",
+    populate: { path: "user", select: "_id id name nickname profileImageUrl" },
+  },
+];
+
+const createHandler = async (req, res) => {
+  try {
+    const { reportedId, type, etcDetail, time } = req.body;
+    const user = await userModel.findOne({ id: req.userId });
+    const creatorId = user._id;
+
+    const reported = await userModel.findById(reportedId);
+    if (!reported) {
+      return res.status(400).json({
+        error: "User/report: no corresponding user",
+      });
+    }
+
+    const report = new reportModel({
+      creatorId: creatorId,
+      reportedId: reportedId,
+      type: type,
+      etcDetail: etcDetail,
+      time: time,
```

Comment:
속성명과 값의 이름이 같을 경우 다음과 같이 생략할 수도 있습니다.
```suggestion
      creatorId,
      reportedId,
      type,
      etcDetail,
      time,
```

Reply:
와 감사합니다!